testing/log/service.go

```diff
@@ -281,20 +283,18 @@ func (s *service) readLogFile(context *endly.Context, source *url.Resource, serv
 	return result, nil
 }
 
-func (s *service) readLogFiles(context *endly.Context, service storage.Service, source *url.Resource, logTypes ...*Type) (TypesMeta, error) {
-	var err error
-	source, err = context.ExpandResource(source)
+func (s *service) readLogFiles(context *endly.Context, fs afs.Service, source *url.Resource, logTypes ...*Type) (TypesMeta, error) {
+	source, storageOptions, err := estorage.GetResourceWithOptions(context, source)
 	if err != nil {
 		return nil, err
 	}
-
 	var response TypesMeta = make(map[string]*TypeMeta)
-	candidates, err := service.List(source.URL)
+	candidates, err := fs.List(context.Background(), source.URL, storageOptions...)
```

Comment:
you can use err, from `source, storageOptions, err := estorage.GetResourceWithOptions(context, source)
	source, err = context.ExpandResource(source)` instead of creating new one